### DISCUSSION TITLE
Request a firmware build - 2.6.0

### DISCUSSION THREAD

**Discussion by pfeerick** [EdgeTX/edgetx#1530] [General]
Ask here if you want/need specific options that are not included in the stock builds. Builds will be uploaded to [here (same link as 'By request' below)](http://files.portabledev.xyz/d/bf957dbe31f04e4a95a4/), so check it out before making a request just in case there is already a custom build with your chosen options...

- [Multilingual builds](http://files.portabledev.xyz/d/a54a4cde2b7f4d388a61/) - zip file per transmitter containing all languages built
- [Multilingual builds - NOHELI](http://files.portabledev.xyz/d/65c7948554c34cae8df4/) - zip file per transmitter containing all languages built
- [By request](http://files.portabledev.xyz/d/bf957dbe31f04e4a95a4/)

**Comment by roman-rangel** [comment]
Hello pfeerick, could you please make a build for TX16S with Bluetooth module support?
Thanks in advance.
Roman.

**Reply by pfeerick** [reply]
Done. Check out the directory link in the initial message. 
`cmake -Wno-dev -DPCB=X10 -DPCBREV=TX16S -DBLUETOOTH=Y ..`

**Reply by roman-rangel** [reply]
Thank you very much Peter!

**Reply by Enggmaug** [reply]
Hello, Exactly what I needed !! I did not put much hope into finding it, but finally stumbed upon this thread. Thanks pfeerick !!
And smaller thanks to dractech for expressing the same need I have.

**Reply by pfeerick** [reply]
You're very welcome. And you'll be even happier soon then, as a custom build for things like bluetooth won't be needed for much longer :)

**Comment by 3DRacer** [comment]
Hi pfeerick,
are there already translations available? I would be interested in a German language build.
Thanks in advance.
3DRacer

**Reply by pfeerick** [reply]
Yes, there are builds for all transmitters in all languages - I just added the links to them in the first post.

**Comment by tbone-rd** [comment]
Hi pfeerick

thank you very much for the long awaited noheli version!
I like the st5font, is it possible to make a NOHELI, SQT5FONT version? (and FLEXR9M if not default)

QX7 ACCST

tbone

**Reply by pfeerick** [reply]
Done :) `cmake -Wno-dev -DPCB=X7 -DHELI=NO -DFONT=SQT5 ..` FLEX _should_ be built in / on by default now...

**Comment by tbone-rd** [comment]
Thank you very much!
Will test it next week

At 08:09 04.02.2022, you wrote:

>Done :) cmake -Wno-dev -DPCB=X7 -DHELI=NO 
>-DFONT=SQT5 .. FLEX should be built in / on by default now...
>
>­
>Reply to this email directly, 
><https://github.com/EdgeTX/edgetx/discussions/1530#discussioncomment-2108928>view 
>it on GitHub, or 
><https://github.com/notifications/unsubscribe-auth/ALS7KWEXQZMUPLS442W2SJTUZN3UJANCNFSM5NFYKGAQ>unsubscribe.
>Triage notifications on the go with GitHub 
>Mobile for 
><https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>iOS 
>or 
><https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>Android. 
>
>You are receiving this because you 
>commented.Message ID: 
***@***.***>

**Comment by EMaximus007** [comment]
Hi pfeerick,
Could you please make a build for the Jumper T18 (v1) with External ACCESS mod?
Thanks

**Reply by pfeerick** [reply]
Sorry for the delay... life and other stuff has a habit of getting in the way. Done, and since I misread that at first, there is also a build for the t16. Check the link above. 

`cmake -Wno-dev -DPCB=X10 -DPCBREV=T18 -DHARDWARE_EXTERNAL_ACCESS_MOD=ON ../`

**Reply by EMaximus007** [reply]
This is great!
Thank you very much  pfeerick.
I will put this to good use.

**Comment by Pacmango** [comment]
HI pfeerick.

I hope you're fine?
Could you create a TX16 french build with bluethoot, no heli, internal multi, lua, flexr9m.
A warm thank you !

**Reply by pfeerick** [reply]
Done. See By request link in first post. Internal multi is standard, and flex is also standard, but is incomplete (see #1187 for details). I wasn't sure if you wanted Lua for the Lua custom scripts or not so there is a build with and one without. 

```
cmake -Wno-dev -DPCB=X10 -DPCBREV=TX16S -DHELI=NO -DBLUETOOTH=Y -DTRANSLATIONS=FR ..
cmake -Wno-dev -DPCB=X10 -DPCBREV=TX16S -DHELI=NO -DBLUETOOTH=Y -DTRANSLATIONS=FR -DLUA_MIXER=Y ..
```

**Reply by Pacmango** [reply]
Wonderfull ! and thank you for the Lua srcipt. i need it !
A great thank you Erick. Good day !

**Comment by nickboul** [comment]
Hello. Great job there. If it is possible I would like a en, Bluetooth, noheli, nogvar build for the tx16s. 

Thank you.

**Reply by pfeerick** [reply]
Sorry for the delay... done. There are builds with and without the lua custom mixer script capability at the link in the first post. 
```
cmake -Wno-dev -DPCB=X10 -DPCBREV=TX16S -DHELI=N -DBLUETOOTH=Y -DGVARS=N -DLUA=Y -DLUA_MIXER=Y -DCMAKE_BUILD_TYPE=Release ..
cmake -Wno-dev -DPCB=X10 -DPCBREV=TX16S -DHELI=N -DBLUETOOTH=Y -DGVARS=N -DLUA=Y -DLUA_MIXER=N -DCMAKE_BUILD_TYPE=Release ..
```

**Reply by nickboul** [reply]
Thank you very much

**Reply by nickboul** [reply]
If i reflash the tx16 wirh the custom fw, will i loose models and widgets?

**Reply by pfeerick** [reply]
No, you should only lose widgets when converting from OTX -> ETX.... after that you should be fine... Upgrading from 2.5 -> 2.6 had some minor hiccups, but usually it was just a matter of tricking the radio in to trying to convert again, or using Companion to import models. Other than that, there should not be any lose of models or settings... but it pays to keep a backup anyway.

**Reply by nickboul** [reply]
I confirm that I did not loose models, widgets or settings.

Thank you once again

**Comment by zhalenik** [comment]
Hello Peter, could you please make a build for t18 with Bluetooth module, serial ports, noheli, and lua support? Thank you!)

**Reply by pfeerick** [reply]
Done. Check the link in the first post. There are builds with and without the lua custom mixer scripts capability. 
```
cmake -Wno-dev -DPCB=X10 -DPCBREV=T18 -DBLUETOOTH=Y -DHELI=N -DLUA=Y -DLUA_MIXER=Y -DCMAKE_BUILD_TYPE=Release ..
cmake -Wno-dev -DPCB=X10 -DPCBREV=T18 -DBLUETOOTH=Y -DHELI=N -DLUA=Y -DLUA_MIXER=N -DCMAKE_BUILD_TYPE=Release ..
```

**Reply by zhalenik** [reply]
Thank you very much!!)

**Reply by zhalenik** [reply]
Did I understand correctly that the Bluetooth mode included in this firmware allows me to use only a special FrSky module (which can be initialized)? Could you explain please... Can I use the HC-05/06 module? There is no signal on the rx tx pins (for the FrSky bt module) now, regardless of which Bluetooth mode I use: trainer(slave) or telemetry. Is this due to the fact that the serial interface of the hc05 module is not initialized as a bluetooth module? Is it possible to support Serial Port(TTL) in the firmware? Thank you in advance!

**Reply by pfeerick** [reply]
If you _just_ want telemetry... it might be worth looking at 2.7, as that will allow runtime configuration of the serial ports, including setting telemetory mirroring. Then you would just want to connect a bluetooth to the serial tx and rx pins, and there would be no 'bluetooth handling' as such by the transmitter. In as far as bluetooth managed by the transmitter... I would have to defer to @dlktdr @rotorman since they know and use the bluetooth stuff (I just know it exists atm :grin:)...

**Reply by zhalenik** [reply]
Thanks a lot for the detailed answer!  It's nice to have support)

**Reply by dlktdr** [reply]
Yes the Bluetooth code in OTX/ETX is specific to the Frsky firmware on the BLE modules. It uses a custom AT command set that isn't compatible with other modules (HC series).

**Reply by zhalenik** [reply]
Thank you!

**Reply by c608345** [reply]
https://openrcforums.com/forum/viewtopic.php?t=7640
This thread says hc-05 works with bluetooth trainer.

**Reply by pfeerick** [reply]
Except, that is ersky9x specific, not edgetx or even opentx ;)
> No need to say, that you will need two radios with a BT module and Ersky9x installed

**Reply by c608345** [reply]
I don't know  whether bluetooth trainer  is hardware independent.

**Reply by pfeerick** [reply]
It's not hardware dependent, it's the instruction set used by the frsky bluetooth trainer module vs that used on the HC-05... So it might be possible to use the HC-05 in the future. I'm not sure if it would be possible to mix and match (i.e. HC-05 on your TX16s and X9D+2019 with frsky bl module), but same module on both sides should be doable.

**Reply by dlktdr** [reply]
The Frysky modules are Bluetooth low energy where as the HC05/06 are BT Classic, which cannot talk to each other. The HM11 is a Bluetooth LE device that might work.. but they use a different service and characteristic to talk. So long story short no none of them are compatible.

Using two HM10/11 or a HC05/06 would be possible had frsky opted for the same AT command set, but they didn't. Some pretty easy code changes to bluetooth.cpp would make this possible.

I started working on a alternative open source one based on the ESP32 as they are cheap and once it's working can use ESPnow for some actual long range trainer/telemetery and BLE or BT classic. If anyone wants to help I would appreciate it! Need to finish the labels selector, then can focus on this again ;) https://github.com/dlktdr/BTWifiModule

**Reply by c608345** [reply]
I found something.

https://www.rcgroups.com/forums/showthread.php?3718549
https://www.rcgroups.com/forums/showpost.php?s=05df21178271a075f7dc5dac696e2682&p=46798539&postcount=7

**Reply by zhalenik** [reply]
@c608345, at the specified link, an example of connecting hc-05 to a tx16s transmitter (not t18) for telemetry out. Probably this method is no longer relevant, because there is an easier way: in the latest versions of the opentx firmware for tx16s (in edgetx maybe too, i don't know), you can choose different modes for two available Serial ports TTL, such as telemetry out/in or sbus trainer and so is LUA. In particular, no settings are even required for telemetry in this case, except to pre-configure baudrate on the hc-05 module. Because hc-05/06 is a "transparent" module, it broadcasts everything that comes to it from the uart. It is important that the hc module is connected not as a bluetooth interface, but as a kind of uart device (it does not interact with the transmitter in any way. work without AT commands, initialization and MAC-address). SO all we need now is the same support for the serial ports in T18 firmware (like in tx16s) because we definitely have a free UART in this transmitter.

[https://www.youtube.com/watch?v=VuPD2GFMVKI&t=148s](url)

[https://www.youtube.com/watch?v=0UulXKpc6Kg](url)
[https://www.youtube.com/watch?v=4lThsp4Q4YA](url)

**Comment by 3DRacer** [comment]
Hi pfeerick,
could you please make a german build for the tx16s with the nice channel preview from PR 1713 on the mixer page? I think its Version 2.7.0 (a7c123b1). The multilanguage versions in the first post are a bit old.

thanks in advance
3DRacer

**Reply by rotorman** [reply]
See https://github.com/EdgeTX/edgetx/wiki/Building-radio-firmware-in-a-webbrowser-with-Gitpod how to easily make your own build in a web-browser, without having to setup/install any software on your PC.

PR #1713 is closed w/o merging, see PR #1748 instead (merged to main).

**Reply by 3DRacer** [reply]
thats great!
Thank you very much for pointing me in the right direction. i will try that immediately.
BR Holger
